### PR TITLE
Uniform Dockerfiles

### PR DESCRIPTION
**Pull request by adamgering**
This promotes volume mounts to the default configuration
and removes the seperate "dev" mode.

This makes all docker files use a base set of common
dependencies to maximize docker image caching.

Renamed tmobile-rbac to hyperledger-rbac as the working
directory name. Run bin/clean to remove pycache directories
that may otherwise cause an error.

bin/clean will also suggest pruning all existing docker
images. That is optional but recommended.

bin/start, bin/start -d and docker-compose up are equivalent.
still use the -b flag as --build or -p flag for persistent
data on bin/start.

Terming into the containers shows the old tmobile working path.
This may be cached somewhere else, cd .. && cd hyperledger-rbac

Issue #291 #353 #354

### CODE REVIEW

**Comment by adamgering** [comment]
Providers should also use the same base docker file config (python 3.5.2), but I left it for now (python 3.7).

I didn't add the new client to the docker-compose yet, as it appears to be in a non-working state.

**Comment by adamgering** [comment]
A few of the files git is showing as renamed + changed are rather deleted files + new files.

File: README.md [hunk 1]
```diff
@@ -174,9 +174,9 @@ hanging in the legacy UI and stack traces from rbac_server:
     rbac-server    |     response = await response
```
Comment (kthblmfld) [inline README.md]: Does this remove the need for the -d option in bin/start? If so, please remove references to it in the README. Also, whatever you can add to README/Wiki to capture this change == Good stuff.

**Comment by jbobo** [comment]
Is there any reason not to just rebuild images by default? What's the use-case where we wouldn't want to?

**Comment by adamgering** [comment]
> Is there any reason not to just rebuild images by default? What's the use-case where we wouldn't want to?

There's no use case where rebuilding the images is beneficial. You mean instead of doing a volume mount?

File: README.md (hunk 1 above)
Comment (adamgering) [reply README.md]: Updated

**Comment by jbobo** [comment]
But rbac-client was added to docker-compose in this PR, right? It's adding a lot of noise to the logs so maybe at the least we should comment it out with a `TODO` and reach out to Phil for more info regarding these errors before we integrate it?

**Comment by kthblmfld** [comment]
@pgobin If you get the chance, please have a look at this PR and help us sort out what is breaking while deploying the client in the fleet of containers. It looks like a React component is missing (jsx-a11y). The logs:

./src/[multipleFiles].js
Line 1:  Definition for rule 'jsx-a11y/href-no-hash' was not found  jsx-a11y/href-no-hash

**Comment by adamgering** [comment]
> @pgobin If you get the chance, please have a look at this PR and help us sort out what is breaking while deploying the client in the fleet of containers. It looks like a React component is missing (jsx-a11y). Missing from package.json?
> 
> The logs:
> 
> ./src/[multipleFiles].js
> Line 1: Definition for rule 'jsx-a11y/href-no-hash' was not found jsx-a11y/href-no-hash

Issue #355 
This is not related to the changes made by this PR, you are seeing those errors only because the client was not previously included in docker-compose.

Approve and merge this PR.

File: docker-compose.yaml [hunk 2]
```diff
@@ -112,14 +120,25 @@ services:
...
-    image: rbac-ui-production:${ISOLATION_ID-latest}
+    image: rbac-ui:${ISOLATION_ID-latest}
     ports:
       - "4200:4200"
     depends_on:
       - rbac-server
     environment:
       - HOST=${HOST:-localhost}
 
+  rbac-client:
```
Comment (kthblmfld) [inline docker-compose.yaml]: This is the change that @jbobo is referring to, and it is part of this PR along with the cleanup of the rbac-client Dockerfile. Both changes are improvements with regards to incorporating the new UI, but the missing dependency causes a dump of errors in the logs upon startup.

**Comment by pgobin** [comment]
@adamgering @kthblmfld @jbobo I'm taking a look at this and #355 now.

**Pull request merged**